spec/system/account_request_system_spec.rb

```diff
@@ -1,44 +1,55 @@
-RSpec.describe "Account request flow", type: :system, js: true do
+RSpec.describe 'Account request flow', type: :system, js: true do
+  it 'should prompt prospective users to request an account on the live app if the new account form is accessed on a staging environment' do
+    visit new_account_request_path
+
+    # Check for staging
+    if Rails.env.development? || Rails.env.production? || Rails.env.test?
+      expect(page).to have_selector(id: 'new_account_request')
+    else
+      expect(page).to have_link('click here', href: 'https://diaper.app/account_requests/new')
+    end
+  end
```

Comment:
I think this is pretty close to how I would do this. However, there is one thing that can be fixed here to ensure we are testing thoroughly this feature. Notably, we will never be running the test in any other environment other than `test` and thus `Rails.env.test?` will always be true in this case. Therefore, the `else` block will never really run. Ideally, we want to build a test that can check both cases in which you are in `staging` or not. The challenge with this that you'll need to inform the test suite that you want to "fake" or "emulate" that the test is running in one environment.

The solution for this would be using a stub to emulate that the test is running in a specified environment and using contexts to separate the two. For example:
```
context 'when in staging' do
    before do
       allow(Rails).to receive_message_chain(:env, :staging?).and_return(true)
    end 

    # add `it` for the case that it is in staging
end

context 'when not in staging' do
    before do
       allow(Rails).to receive_message_chain(:env, :staging?).and_return(false)
    end 

    # add `it` for the case that it is not in staging
end
```

This mocking is really powerful in these cases. You can read more here - https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/message-chains

Let me know if you have any Qs via slack if you want to dig more into this.